app/assets/javascripts/new_layout.js

```diff
@@ -0,0 +1,27 @@
+window.ST = window.ST || {};
+
+(function(module) {
+
+  function check_handler(element) {
+   return function(){
+      if(!element.prop("checked")){
+        element.prop("checked", this.checked);
+      }
+      element.attr("disabled", this.checked)
+    }
+  };
+
+  // Disables and enables required checkboxes for parent flags
+  module.initializeNewLayoutManager = function(feature_rels){
+    Object.keys(feature_rels).forEach(function(key,index) {
+      var $parent_for_user = $("#enabled_for_user_" + key);
```

Comment:
I guess, this is OKish. However, I have a habit of adding className of parent container to params, so that these could be restricted to certain part of the page (you never know if there is more components with same name added in the future).

That way you could write the content in a vanilla js manner:
```javascript
var parent_for_user = document.querySelector(parentClass + " #enabled_for_user_" + key);
var required_for_user =  document.querySelector(parentClass + " #enabled_for_user_" + feature_rels[key]);
parent_for_user.addEventListener("click", check_handler(required_for_user));
```